src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Manufacturer;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return getAll().stream()
+                .filter(manufacturer -> Objects.equals(manufacturer.getId(),id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturerStorage;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer oldManufacturer = get(manufacturer.getId()).get();
+        oldManufacturer.setCountry(manufacturer.getCountry());
+        oldManufacturer.setName(manufacturer.getName());
```

Comment:
this code isn't flexible. We should change the logic of this method if the manufacturer object is changed
let's use the method `set` from the list for replacing an object

Reply:
Fixed